guides/operator/custom-job-stages/index.md

```diff
@@ -0,0 +1,122 @@
+* `cloudProvider` - Cloud provider to use for this stage. Valid values are `titus` or `kubernetes`.
+* `credentials` - Account name used when executing the job. Must be valid account for `cloudProvider`.
+* `waitForCompletion` - If `true`, the stage will not be marked complete until the job completes.
+* `parameters` - List of parameters presented to the end user.
+
+### Custom Job Stages - Kubernetes
+
+The following properties are supported for the `kubernetes` cloud provider.
+
+* `account` - Account name used when executing the job. Must be valid account for `cloudProvider`.
+* `application` - Name of the application to associate the job with.
+* `manifest` - YAML definition for the [Kubernetes Job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) that will be run when the stage is executed.
+
+### Custom Job Stages - Titus
+
+* `cluster` - Definition of the cluster to be created when the stage is executed.
+* `region` - Region in which to run the underlying job.
+* `propertyFile` - File to download once the job completes.
+
+## Custom Job Stage parameters
+
+Using parameters with the Custom Job Stage enables your users to customize specific fields of the job configuration at runtime. Before the stage is started, user supplied values will replace the values in the stage's configuration. Parameters support the following properties
+
+* `name` - Name of parameter.
+* `label` - Human readable parameter name (shown in the UI).
+* `defaultValue` - default value, if not overridden by user.
+* `description` - Description of what parameter is used for.
+* `order` - Display order of the parameter in the UI.
+
+
+
+## Example Custom Job Stage
+
+```
+job:
+  preconfigured:
+    kubernetes:
+      - label: Echo
+        type: echoPhrase
+        description: Echo a configured phrase
+        cloudProvider: kubernetes
+        account: kubernetes
+        credentials: kubernetes
+        waitForCompletion: true
+        application: k8s2
+        parameters:
+          - name: PHRASE
+            description: Phrase to be echoed.
+            mapping: manifest.spec.template.spec.containers[0].env[0].value
+            defaultValue: "Hello world!"
+        manifest:
+          apiVersion: batch/v1
+          kind: Job
+          metadata:
+            name: say-hello
+            namespace: spinnaker
+          spec:
+            backoffLimit: 0
+            template:
+              spec:
+                restartPolicy: Never
+                containers: 
+                  - name: hello
+                    image: alpine:latest
+                    env:
+                      - name: PHRASE
+                        value: placeholder
+                    command: ["echo", $(PHRASE)]
+```
+
+## Using a Custom Job Stage
+
+Custom Job Stages will appear as native stages within the Spinnaker UI. Users can add these stages to their pipelines as they would any other stage. 
```

Comment:
@ethanfrogers Could/should you add this to /reference/pipeline/stages/ ?

Reply:
+1. That should help with discovery.

Reply:
@dorbin take a look and LMK what you think.